components/job-orchestration/job_orchestration/scheduler/search/search_scheduler.py

```diff
@@ -110,20 +110,29 @@ def get_archives_for_search(
     db_conn,
     search_config: SearchConfig,
 ):
-    query = f"""SELECT id as archive_id
-            FROM {CLP_METADATA_TABLE_PREFIX}archives
-            """
+    query = f"SELECT id as archive_id FROM {CLP_METADATA_TABLE_PREFIX}archives"
     filter_clauses = []
     if search_config.end_timestamp is not None:
         filter_clauses.append(f"begin_timestamp <= {search_config.end_timestamp}")
     if search_config.begin_timestamp is not None:
         filter_clauses.append(f"end_timestamp >= {search_config.begin_timestamp}")
+    if search_config.tags is not None:
+        filter_clauses.append(
+            f"id IN (SELECT archive_id FROM {CLP_METADATA_TABLE_PREFIX}archive_tags WHERE "
```

Comment:
Can we add a comment about the reasoning for using nested subqueries rather than a table join? I guess it's because we expect many more archives than tags, so a join may be inefficient but then again, I'm not sure how MySQL would optimize this.

Reply:
Generally speaking, join is faster than subqueries. But in our case, the performance is similar according to the query plan. And it's easier to write the code.

Reply:
Do you think I should change subqueries(or the second level) to join?

Reply:
If the query plan is similar, I think the current way of writing it makes sense.